From b7b8f10a4b31a879bfee00b948160747a30c6aad Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@opensource.altera.com>
Date: Mon, 21 Mar 2016 16:21:37 -0500
Subject: [PATCH 085/142] FogBugz #366816-1: Fix up Denali NAND for v4.5

commit "442f201b93b5 mtd: nand: denali: use the mtd instance embedded in
struct nand_chip" removes the need to use the denali structure since
the struct nand_chip now embeds an mtd device.

Signed-off-by: Dinh Nguyen <dinguyen@opensource.altera.com>
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/mtd/nand/denali.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/mtd/nand/denali.c b/drivers/mtd/nand/denali.c
index 4e656538d272..cc517eaab521 100644
--- a/drivers/mtd/nand/denali.c
+++ b/drivers/mtd/nand/denali.c
@@ -1224,19 +1224,19 @@ static int denali_read_page(struct mtd_info *mtd, struct nand_chip *chip,
 			/* When we have hw ecc fixup, don't check oob.
 			 * That code below looks jacked up anyway.  I mean,
 			 * look at it, wtf? */
-			if (!is_erased(buf, denali->mtd.writesize))
-				denali->mtd.ecc_stats.failed++;
+			if (!is_erased(buf, mtd->writesize))
+				mtd->ecc_stats.failed++;
 		} else {
-			read_oob_data(&denali->mtd, chip->oob_poi,
+			read_oob_data(mtd, chip->oob_poi,
 				denali->page);
 
 			/* check ECC failures that may have occurred on
 			 * erased pages */
 			if (check_erased_page) {
-				if (!is_erased(buf, denali->mtd.writesize))
-					denali->mtd.ecc_stats.failed++;
-				if (!is_erased(buf, denali->mtd.oobsize))
-					denali->mtd.ecc_stats.failed++;
+				if (!is_erased(buf, mtd->writesize))
+					mtd->ecc_stats.failed++;
+				if (!is_erased(buf, mtd->oobsize))
+					mtd->ecc_stats.failed++;
 			}
 		}
 	}
@@ -1705,14 +1705,14 @@ int denali_init(struct denali_nand_info *denali)
 				MAIN_ACCESS);
 	}
 
-	if (nand_scan_tail(&denali->mtd)) {
+	if (nand_scan_tail(mtd)) {
 		ret = -ENXIO;
 		goto failed_req_irq;
 	}
 
 	/* We use the parse function and pass the of_node bcs
 	   we want to pick up partitions from device tree */
-	ret = mtd_device_parse_register(&denali->mtd, NULL,
+	ret = mtd_device_parse_register(mtd, NULL,
 			&(struct mtd_part_parser_data){
 				.of_node = denali->dev->of_node,
 			},
-- 
2.11.0

